http4s/client/jvm/src/test/scala/endpoints4s/http4s/client/Http4sClientUrlEncodingTest.scala

```diff
@@ -0,0 +1,40 @@
+package endpoints4s.http4s.client
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import endpoints4s.algebra
+import org.http4s.Uri
+import org.http4s.asynchttpclient.client.AsyncHttpClient
+import org.scalatest.BeforeAndAfterAll
+
+class Http4sClientUrlEncodingTest
```

Comment:
We might want to consider defining a base class for the JVM and JS test suites, to avoid code duplication between them.

Reply:
I thought about leaving this test in `shared`, but for some reason SBT was ignoring it. I think I had some weird caching issues from mixed use of SBT directly and through BSP, I will try it again.

Reply:
I wonder how to set it up in sbt - `http4s-client` `shared` has to depend on algebra, which is a cross project in itself. Any ideas?

`.dependsOn(algebra % "test->test;compile->compile")`
Just throws errors:
```
found   : sbt.ClasspathDependency
 required: sbtcrossproject.CrossClasspathDependency
    .dependsOn(algebra % "test->test;compile->compile")
```

Reply:
Figured it out!
```
.dependsOnLocalCrossProjectsWithScope(
      "algebra" -> "test->test;compile->compile"
    )
```

Reply:
Indeed!

I plan to move the “test” part of the algebra to a separate project soon, so that we can just `dependsOn` it.